### PR TITLE
Add support for resampling-filter option

### LINKED ISSUE #107: Improve sharpness of resized images
It seems that images that are resized by the browser appear sharper compared to images resized by Imagine (tested with JPEG quality 100). We should try to find out how browsers achieve that and improve the resize process to get similar results.

### PR DESCRIPTION
**Pull request by ausi**
Fixes #107

### CODE REVIEW

File: src/Resizer.php [hunk 1]
```diff
@@ -161,8 +164,15 @@ protected function executeResize(ImageInterface $image, ResizeCoordinates $coord
             (new Autorotate())->apply($imagineImage);
         }
 
+        if ($imagineImage instanceof SvgImage || $imagineImage instanceof GdImage) {
+            // Backwards compatibility with imagine/imagine <= 1.2.4 and contao/imagine-svg <= 1.0.3
+            $filter = ImagineImageInterface::FILTER_UNDEFINED;
+        } else {
+            $filter = $imagineOptions['resampling-filter'] ?? ImagineImageInterface::FILTER_LANCZOS;
```
Comment (ausi) [inline src/Resizer.php:171]: The naming using a dash seems inconsistent, but this option exists exactly like that already in Imagine so I used the same name: https://github.com/php-imagine/Imagine/blob/93bdd9ad420bfbf2aa5c33fe70d134b3c9cc081b/src/Imagick/Image.php#L870

Comment (Toflar) [reply src/Resizer.php:171]: Did you see the comment on https://www.php.net/manual/de/imagick.resizeimage.php#94493

> I ended up choosing CATROM as it has a very similar result to LANCZOS, but is significantly faster.

It's from 14 years ago but maybe the algorithm speed difference still applies?

Comment (Toflar) [reply src/Resizer.php:171]: I just read up on those two algorithms and they differ very much in their approach so I don't think that they would have "similar results" 😏

Comment (ausi) [reply src/Resizer.php:171]: Yes, hard to trust any statement regarding that topic as many people mean totally different things when they talk about sharpness and quality of resized images :/

**Review by leofeyer — APPROVED** [review]

**Review by fritzmg — APPROVED** [review]

**Pull request merged**